lib/kubernetes-deploy/kubernetes_resource/deployment.rb

```diff
@@ -123,5 +157,20 @@ def find_latest_rs(deployment_data)
       rs.sync(latest_rs_data)
       rs
     end
+
+    def min_available_replicas
+      max_unavailable = @definition.dig('spec', 'strategy', 'rollingUpdate', 'maxUnavailable')
+      if max_unavailable =~ /%/
+        (@desired_replicas * (100 - max_unavailable.to_i) / 100.0).ceil
+      else
+        @desired_replicas - max_unavailable.to_i
+      end
+    end
+
+    def required_rollout
```

Comment:
I put this back because `validate_definition` gets called before `sync` and I wasn't happy extracting the definition value there.

Reply:
Wait a second, why don't we just use the value we see in the definition all the time? This annotation isn't something that'd be server-side defaulted.